backend/inspirehep/records/api/mixins.py

```diff
@@ -326,11 +328,24 @@ def get_newest_linked_conferences_uuid(self):
 
 
 class InstitutionPapersMixin:
+    institution_fields = [
+        "authors.affiliations.record",
+        "thesis_info.institutions.record",
+        "record_affiliations.record",
+    ]
+
     def clean_institution_literature_relations(self):
         InstitutionLiterature.query.filter_by(literature_uuid=self.id).delete()
 
     def create_institution_relations(self):
-        institutions = self.get_linked_records_from_field("authors.affiliations.record")
+        institutions = list(
```

Comment:
why not a `set`? also you seem to be using this several times. Wouldn't it be better to have a `linked_institutions` property on the record?

Reply:
I needed in some place the record and in others the pids. But then I realized that we can have the same institution linked in 2 different fields in a record so I got pids for here too to apply a set and then retrieved the records.